service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java

```diff
@@ -1445,24 +1465,48 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
       IcebergTableLikeEntity entity =
           IcebergTableLikeEntity.of(resolvedPath == null ? null : resolvedPath.getRawLeafEntity());
       String existingLocation;
+      int maxMetadataCacheBytes =
+          callContext
+              .getPolarisCallContext()
+              .getConfigurationStore()
+              .getConfiguration(
+                  callContext.getPolarisCallContext(),
+                  catalogEntity,
+                  FeatureConfiguration.METADATA_CACHE_MAX_BYTES);
+      Optional<String> metadataJsonToCache =
+          switch (maxMetadataCacheBytes) {
+            case FeatureConfiguration.METADATA_CACHE_MAX_BYTES_NO_CACHING -> Optional.empty();
+            case FeatureConfiguration.METADATA_CACHE_MAX_BYTES_INFINITE_CACHING -> {
+              yield Optional.of(TableMetadataParser.toJson(metadata));
+            }
+            default -> {
+              String rawMetadataJson = TableMetadataParser.toJson(metadata);
+              if (rawMetadataJson.length() < maxMetadataCacheBytes) {
```

Comment:
Not quite accurate, we would be underestimating a bit here potentially by a factor of 4. We would have to use bytes().length to actually get the size here. I think just multiplying by 2 is probably a good conservative upper bound down if we don't want to call bytes.